### PR TITLE
fix: Add dependency on instructlab-training for ROCm

### PR DESCRIPTION
**Pull request by fabiendupont**
**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: requirements/rocm.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 # Extra dependencies for AMD ROCm
+instructlab-training[rocm]>=0.5.0
```
Comment (nathan-weinberg) [inline requirements/rocm.txt:2]: Just as a heads up, this change will require a new release of the training library (`0.5.3` I believe) once https://github.com/instructlab/training/pull/255 merges

@fabiendupont once this PR is merged to `main` we will need to backport it to `release-v0.19` - I can assist with that

**Review by nathan-weinberg — APPROVED** [review]

**Comment by nathan-weinberg** [comment]
@Mergifyio backport release-v0.19

File: requirements/rocm.txt (hunk 1 above)
Comment (tiran) [reply requirements/rocm.txt:2]: We need to wait until https://github.com/instructlab/training/pull/260 has landed and another release of instructlab-training is available.

Comment (tiran) [reply requirements/rocm.txt:2]: An hour ago, the dependency would have failed. 0.5.3 and earlier do not have a `rocm` extra. 0.5.4 added the extra.

**Review by tiran — APPROVED** [review]
https://pypi.org/project/instructlab-training/0.5.4/ with `Provides-Extra: cuda, rocm` is available.

File: requirements/rocm.txt (hunk 1 above)
Comment (nathan-weinberg) [reply requirements/rocm.txt:2]: No building will occur until `0.19.3.` is released

**Comment by tiran** [comment]
@nathan-weinberg Mergio isn't merging the PR. Could you take a look, please?

**Comment by nathan-weinberg** [comment]
> @nathan-weinberg Mergio isn't merging the PR. Could you take a look, please?

Yes, I'm working on a fix right now. Will merge this manually in the meantime.

**Pull request merged**